projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py

```diff
@@ -74,8 +74,27 @@ def run(job_input)
             sys.path.insert(0, str(step.job_dir))
             filename = step.file_path.name
             namespace = "step_" + str(filename).lower().strip(".py")
-            python_module = imp.load_source(namespace, str(step.file_path))
             success = False
+
+            try:
+                log.debug("Loading %s ..." % filename)
+                python_module = imp.load_source(namespace, str(step.file_path))
+                log.debug("Loading %s SUCCESS" % filename)
+            except BaseException as e:
```

Comment:
Is there a specific reason for catching BaseException and not just Exception?

Reply:
saw same approach in `cli_run.py` and `data_job.py` on the way the exception propagates

Reply:
Maybe we need to consider how we construct exceptions. According to the documentation, BaseException is not meant to be inherited from https://docs.python.org/3/library/exceptions.html#BaseException, so in theory, we shouldn't expect to need to catch any exception at that low level